src/lib/containers/QueryWrapper.tsx

```diff
@@ -37,6 +37,8 @@ export type QueryWrapperProps = {
   showBarChart?: boolean
   componentIndex?: number //used for deep linking
   shouldDeepLink?: boolean
+  onQueryChange?: (newQueryJson: string) => void
+  onQueryResultBundleChange?: (newQueryResultBundleJson: string) => void
```

Comment:
Provide way for caller to listen for QueryResultBundle changes (SWC uses this extensively to figure out state of other UI widgets).